scripts/set_pass

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+# These give serial numbers, but serials are relatively predictable
+# so let's not use them to generate a password.
+#cat /proc/cpuinfo | sed -n 's/Serial[[:space:]]*:[[:space:]]0*//p'
+#cat /sys/block/mmcblk0/device/cid
+
+# Password cracking speed: ~2 tps (online, connect via ssh and attempt login)
+#                          ~3 MH/s with 8x NVIDIA RTX 3080s (somehow got hash/salt offline)
+# [linux uses sha512crypt by default: SHA512 with 5000 rounds]
+# https://gist.github.com/Chick3nman/bb22b28ec4ddec0cb5f59df97c994db4
+# Test password cracking speed over SSH to get 'online' tps
+#echo "pi" > users.txt
+#xkcdpass -c 100 -n 4 --min 3 --max 7 -d '' -C first > passwords.txt
+#nmap amplipi.local -p 22 --script ssh-brute --script-args userdb=users.txt,passdb=passwords.txt
+#
+# Number of bits desired = Cracking bits/year + 1 bit for average
+#                        = log2(3*10^6*3600*24*365) + 1 = 47.4 bits
+
+# If AmpliPi is opened to the internet or in general additional security is required,
+# key-based authentication should be used. See
+# https://www.raspberrypi.com/documentation/computers/configuration.html#using-key-based-authentication
+
+set -e
+confdir="$HOME/.config/amplipi"
+
+# First verify the current user is 'pi'
+#if [[ $USER != pi ]]; then
+#  echo "ERROR: Not running as pi user."
+#  exit 1
+#fi
+
+HELP="Set a password on AmpliPi\n
+usage: set_pass [PASSWORD]\n
+If no PASSWORD is given, generate a random passphrase,\n
+otherwise set the pi user's password to PASSWORD.\n"
+
+pass=''
+while [ "$#" -gt 0 ]; do
+  case $1 in
+    -h|--help) echo -e $HELP; exit 0 ;;
+    *) pass=$1 ;;
+  esac
+  shift
+done
+
+# Next check if a password has already been generated
+#if [[ -f $confdir/default_password.txt &&  ]]; then
+#fi
+
+if [ -z "$pass" ]; then
+  # Generate a random passphrase and save to file.
```

Comment:
Would it make sense to make a function for each of these different password options?